### PR TITLE
[air] add feast example

### PR DESCRIPTION
**Pull request by xwjiang2010**
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by xwjiang2010** [comment]
A bit clueless of the error "feast_example.ipynb:210003: WARNING: 'myst' reference target not found:"
Do you guys know how to fix it by any chance?

File: doc/source/ray-air/examples/tfx_tabular_train_to_serve.ipynb [hunk 1]
```diff
@@ -6,6 +6,8 @@
     "id": "VaFMt6AIhYbK"
    },
    "source": [
+    "# Tabular data training and serving using Ray Air\n",
```
Comment (xwjiang2010) [inline doc/source/ray-air/examples/tfx_tabular_train_to_serve.ipynb:9]: The index page only renders with the first level 1 header in the notebook. Currently it shows "set up ray"

Comment (richardliaw) [reply doc/source/ray-air/examples/tfx_tabular_train_to_serve.ipynb:9]: nice :)

**Comment by richardliaw** [comment]
BTW, my understanding is that you want to use the feature store during online inference, and the client usually doesn't have access to the feature store (only the inference server does).

In this example, it seems like features are pulled on the client side. Maybe we can put together an example for online inference?

**Comment by xwjiang2010** [comment]
> BTW, my understanding is that you want to use the feature store during online inference, and the client usually doesn't have access to the feature store (only the inference server does).
> 
> In this example, it seems like features are pulled on the client side. Maybe we can put together an example for online inference?

Yeah that's a good point. In that case, we should design a component to do something similar as a [Transformer](https://kserve.github.io/website/master/modelserving/v1beta1/transformer/feast/) in KServe's terminology. Probably some augmentation to ModelWrapper. cc @simon-mo @edoakes to see what they think.

**Comment by xwjiang2010** [comment]
Or maybe a dedicated Preprocessor.

**Comment by simon-mo** [comment]
Preprocessor makes the most sense here!

**Comment by xwjiang2010** [comment]
Sounds good. Can we merge this example in and follow up with feast integration in preprocessors as a P1?

File: doc/source/ray-air/examples/feast_example.ipynb [hunk 2]
```diff
@@ -0,0 +1,1512 @@
...
+   "metadata": {
+    "id": "DcPIskZlzSal"
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "WORKING_DIR = os.environ.get(\n",
+    "    \"WORKING_DIR\", os.path.expanduser(\"~/ray-air-feast-example/\")\n",
+    ")\n",
+    "os.environ[\"WORKING_DIR\"] = WORKING_DIR"
```
Comment (richardliaw) [inline doc/source/ray-air/examples/feast_example.ipynb:51]: Hmm, why do we have to do this?

Comment (xwjiang2010) [reply doc/source/ray-air/examples/feast_example.ipynb:51]: wrong link?

Comment (richardliaw) [reply doc/source/ray-air/examples/feast_example.ipynb:51]: updated https://stackoverflow.com/a/40932507/14961124

Comment (xwjiang2010) [reply doc/source/ray-air/examples/feast_example.ipynb]: nice! updated!

**Review by richardliaw — APPROVED** [review]

**Pull request merged**